File: main/coreplugins/io.sarl.intellij/src/main/java/io/sarl/intellij/SarlModuleType.java
package io.sarl.intellij;

import com.intellij.ide.util.projectWizard.ModuleBuilder;
import com.intellij.openapi.module.ModuleType;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class SarlModuleType extends ModuleType {

    public static final SarlModuleType INSTANCE = new SarlModuleType();

    protected SarlModuleType() {
        super("sarl");
    }

    @NotNull
    @Override
    public ModuleBuilder createModuleBuilder() {
        return SarlModuleBuilder.INSTANCE;
    }

    @NotNull
    @Override
    public String getName() {
        return "Sarl";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Sarl module";
    }

    @Override
    public Icon getNodeIcon(boolean isOpened) {
        return SarlIcons.SARL_PLUGIN;
    }
}

File: main/coreplugins/io.sarl.intellij/src/main/kotlin/io/sarl/intellij/SarlModuleType.kt
package io.sarl.intellij

import com.intellij.openapi.module.ModuleType
import com.intellij.openapi.module.ModuleTypeManager
import javax.swing.Icon

class SarlModuleType : ModuleType<SarlModuleBuilder>(ID) {
    override fun createModuleBuilder(): SarlModuleBuilder {
        return SarlModuleBuilder.INSTANCE
    }

    override fun getName(): String {
        return "Sarl"
    }

    override fun getDescription(): String {
        return "Sarl module"
    }

    override fun getNodeIcon(isOpened: Boolean): Icon {
        return SarlIcons.SARL_PLUGIN
    }

    companion object {
        private const val ID = "SARL_MODULE"
        val INSTANCE: SarlModuleType by lazy { ModuleTypeManager.getInstance().findByID(ID) as SarlModuleType }
    }
}
